simfection/interaction_engine.py

```diff
@@ -152,24 +161,92 @@ def _interact(self, a, b):
             )
 
     def interact_all(self):
+        cpp = self.settings.get_setting('cpp')
         verbose = self.verbose
-        logger.debug('- Getting unique interactions.')
-        self._get_unique_connections()
-        connections = self.connections
-        logger.debug('- Running all unique interactions.')
-        for a, b in connections.interaction.values:
-            self._interact(a, b)
-
-        if verbose:
-            print(self.population.state.value_counts())
+        if cpp:
+            logger.debug('- Running InteractionEngine with C++ optimizations.')
+
+            # Initialize the network.PyInteractions object
+            # For a look at the API, checkout the network.pyx file in /cpp_src/cython
+            interactions = network.PyInteractions()
+
+            # Get all the population information required for running PyInteractions
+            pop_agent = self.population['agent'].tolist()
```

Comment:
If we are going to have parallel versions of things, we want to maintain the same namespace, as much as possible. I've started the full naming convention. Essentially, don't abbreviate anything. The prioritization is readability of speed of development (wrt typing). 

Also, the main method (here `interact_all`) should be free of all tasks except writing to logger and calling helper methods. Each section you have should be it's own helper method. See how I implemented the non-cpp version.